### PR TITLE
test(macros): add compile-pass coverage for Depends<FactoryType> in #[get]/#[server_fn]

### LINKED ISSUE #3727: test(macros): add compile-pass coverage for Depends<FactoryType> in #[get]/#[server_fn]
## Summary

PR #3725 fixes the macro codegen so `#[inject] cfg: Depends<T>` works when `T`
is produced by `#[injectable_factory]` (i.e. no `impl Injectable` exists). The
fix routes `Depends<T>` parameters through `resolve_from_registry()`, which
does not require the `T: Injectable` trait bound.

Copilot review on #3725 (threads on `crates/reinhardt-core/macros/src/routes.rs:444`
and `crates/reinhardt-pages/macros/src/server_fn.rs:803`) flagged this as a
regression-prone compilation edge case and suggested adding compile-pass
tests for:

1. `#[get]` with `#[inject] cfg: Depends<FactoryType>` where `FactoryType` is
   registered via `#[injectable_factory]`.
2. `#[server_fn]` with the same pattern.

## Why this is filed as a follow-up

`crates/reinhardt-core/macros/tests/ui.rs` explicitly states:

> Tests avoid depending on other workspace crates (reinhardt-db, reinhardt-di,
> reinhardt-admin, etc.) to avoid circular dependencies during crates.io
> publishing. [...] Full integration testing happens in `tests/integration/`.

So the compile-pass tests cannot live in the macro crate itself and must be
added to `tests/integration/tests/di/ui/pass/` (or a sibling location) where
`reinhardt-di` is available.

## Tasks

- [ ] Add `tests/integration/tests/di/ui/pass/factory_depends_in_route_macro.rs`
      combining `#[injectable_factory]`, `Depends<FactoryType>`, and `#[get]`.
- [ ] Add `tests/integration/tests/di/ui/pass/factory_depends_in_server_fn.rs`
      combining `#[injectable_factory]`, `Depends<FactoryType>`, and
      `#[server_fn]`.
- [ ] Register both test files in `tests/integration/tests/di/ui.rs`.

## Acceptance

Both new compile-pass tests must pass on CI. A regression that reintroduces
the `T: Injectable` bound on the `Depends<T>` path must cause these tests to
fail.

## References

- PR #3725 (the fix)
- Issue #3723 (the bug this fix resolves)
- Copilot review comments:
  - https://github.com/kent8192/reinhardt-web/pull/3725#discussion_r3097245918
  - https://github.com/kent8192/reinhardt-web/pull/3725#discussion_r3097245930

## Labels

`agent-suspect` is applied because this issue is agent-detected and needs
independent verification before being acted on.

### PR DESCRIPTION
**Pull request by kent8192**
## Summary

This PR addresses:

- Add two trybuild compile-pass tests locking in the fix shipped in PR #3725 (bug #3723) covering `#[injectable_factory]`-registered types consumed via `Depends<T>` in `#[get]` and `#[server_fn]` codegen paths
  - `tests/integration/tests/di/ui/pass/factory_depends_in_route_macro.rs`
  - `tests/integration/tests/di/ui/pass/factory_depends_in_server_fn.rs`
- Enable the `"pages"` feature on the `reinhardt` dev-dependency of the integration-tests crate so `#[server_fn]` macro's crate-path resolver can resolve generated paths

## Type of Change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Refactoring (code change that neither fixes a bug nor adds a feature)
- [ ] Documentation update
- [ ] Performance improvement
- [x] Code quality improvements
- [ ] CI/CD changes
- [ ] Other (please describe):

## Motivation and Context

Copilot review on #3725 flagged this compile-time edge case as regression-prone and requested dedicated trybuild coverage. A regression that reintroduces the `T: Injectable` bound on the `Depends<T>` resolution path would cause these tests to fail to compile.

The macro crates themselves cannot host the tests because their `tests/ui.rs` explicitly forbids cross-crate dev-dependencies (to avoid circular-dep issues during crates.io publishing).

Issue #3727 was independently verified before implementation: the `agent-suspect` label was removed after an independent Explore agent confirmed all 4 claims in the issue.

Fixes #3727

Related to: #3725, #3723

## How Was This Tested?

- `cargo test -p reinhardt-integration-tests --test di test_injectable_compile_pass_cases` — 7/7 pass cases (5 existing + 2 new) compile and pass
- `cargo make fmt-check` — 0 would be formatted
- No new Clippy warnings introduced

## Performance Impact

No performance impact — test-only additions.

## Checklist

- [ ] I have followed the [Contributing Guidelines](../blob/main/CONTRIBUTING.md)
- [ ] I have followed the [Commit Guidelines](../blob/main/instructions/COMMIT_GUIDELINE.md)
- [ ] I have updated the documentation (if applicable)
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] I have tested with all affected database backends (if applicable)
- [x] I have formatted the code with `cargo make fmt-fix`
- [x] I have checked the code with `cargo make clippy-check`
- [ ] I use self-hosted runner for CI (Repository owner only)

## Related Issues

- Fixes #3727
- Refs #3725
- Refs #3723

### Type Label (select one)
- [ ] `bug` - Bug fix
- [ ] `enhancement` - New feature or improvement
- [ ] `documentation` - Documentation update
- [ ] `performance` - Performance improvement
- [ ] `refactoring` - Code refactoring
- [x] `code-quality` - Code quality improvements
- [ ] `breaking-change` - Breaking change (also select in "Type of Change" above)

### Scope Label (select all that apply)
- [ ] `database` - Database layer, schema, migrations
- [ ] `auth` - Authentication, authorization, sessions
- [ ] `orm` - ORM layer, models, query builder
- [ ] `http` - HTTP layer, handlers, middleware
- [ ] `routing` - URL routing, path matching
- [ ] `api` - REST API, serializers, views
- [ ] `admin` - Admin interface, admin panels
- [ ] `forms` - Form handling, validation, rendering
- [ ] `graphql` - GraphQL schema, resolvers
- [ ] `websockets` - WebSocket connections, handlers
- [ ] `i18n` - Internationalization, localization
- [ ] `ci-cd` - CI/CD workflow changes

### Priority Label (for maintainers)
- [ ] `critical` - Blocks release or major functionality
- [ ] `high` - Important fix or feature
- [x] `medium` - Normal priority
- [ ] `low` - Minor fix or enhancement

---

**Additional Context:**

Pre-existing Clippy error in `tests/views/views_property_based_integration.rs:202` (`3.14 ≈ PI`) exists on the base commit and is out of scope for this PR.

### CODE REVIEW

File: tests/integration/Cargo.toml [hunk 1]
```diff
@@ -10,7 +10,7 @@ autotests = true
 # Main reinhardt facade crate for proc macro path resolution.
 # Proc macros generate code with absolute paths like ::reinhardt::db::orm::Model
 # This dependency makes those paths resolvable in test files.
```
[the remark this answers is not in the document]
Comment (kent8192) [reply tests/integration/Cargo.toml]: Addressed in 9e947949: expanded the comment above the `reinhardt` dev-dependency in `tests/integration/Cargo.toml` to explicitly state that the `pages` feature is what makes `::reinhardt::pages::...` paths (emitted by `#[server_fn]`) resolvable in trybuild UI tests, alongside the existing `::reinhardt::db::...` rationale.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

File: tests/integration/tests/di/ui/pass/factory_depends_in_server_fn.rs [hunk 2]
```diff
@@ -0,0 +1,32 @@
+//! Compile-pass test: `Depends<FactoryType>` in `#[server_fn]`.
+//!
+//! Regression guard for issue #3723 (fixed in PR #3725). The
+//! `#[server_fn]` macro routes `Depends<T>` parameters through
+//! `resolve_from_registry()`, which has no `T: Injectable` bound. This
+//! allows factory-produced types registered via `#[injectable_factory]`
+//! to be injected without a manual `Injectable` implementation. A
+//! regression that reintroduces the `Injectable` bound on the
+//! `Depends<T>` codegen path would cause this test to fail to compile.
```
[the remark this answers is not in the document]
Comment (kent8192) [reply tests/integration/tests/di/ui/pass/factory_depends_in_server_fn.rs]: Addressed in 9e947949: the test header now references both issues — #3723 as the underlying bug (fixed in PR #3725) and #3727 as the tracking issue for this regression test coverage.

🤖 Generated with [Claude Code](https://claude.com/claude-code)
